### PR TITLE
adding genconfig tests

### PR DESCRIPTION
**Pull request by WendelHime**
This pull request add tests to the genconfig package and check if the provided values are loaded and being written as expected. A bug was found while trying to load the recently added `VerifyHostname`, since this is a optional field that will only appear when the value is defined at the `provider_map.yaml`, the YAML parser tries to parse the value from the provider_map, if it exists it writes the field as a nil value, which is incorrect. This PR add the struct tag with the `omitempty` parameter so the parser works as expected.

### CODE REVIEW

**Comment by hwh33** [comment]
Hey I'm sorry, I believe I have misunderstood the meaning of `defaultProviderID`.  More details [here](https://github.com/getlantern/lantern-cloud/pull/980#discussion_r1733293125), but I think we should probably close this PR (unless I'm still misunderstanding!).  Sorry to have led you astray =/

**Comment by WendelHime** [comment]
Wait, can we leave the tests?

**Comment by hwh33** [comment]
> Wait, can we leave the tests?

Oh, yes please!  Let me take a closer look tomorrow

File: config/client_config.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type ProviderConfig struct {
 	Validator           *ValidatorConfig
 	PassthroughPatterns []string
 	FrontingSNIs        map[string]*fronted.SNIConfig
-	VerifyHostname      *string
+	VerifyHostname      *string `yaml:"verifyHostname,omitempty"`
```
Comment (hwh33) [inline config/client_config.go:46]: Adding context from Slack for posterity: @WendelHime added tests, originally as part of an effort to make a change to the default provider ID.  Though we did not end up changing the default provider ID, Wendel's tests caught a bug here.  The issue is that the YAML parser will always set this value to nil, unless `omitempty` is configured.  This is true even if `verifyHostname` is present in the YAML.  The bug does not present itself currently, but will if we try to use `genconfig` with this new field.

Is this correct, @WendelHime?

**Review by hwh33 — APPROVED** [review]
One comment for clarification, but LGTM!

File: config/client_config.go (hunk 1 above)
Comment (WendelHime) [reply config/client_config.go:46]: Yes, that's it!

**Pull request merged**